### PR TITLE
Fixed #31620 -- Added support for %V format to WeekMixin/WeekArchiveView.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/31620

### CODE REVIEW

File: tests/generic_views/test_dates.py [hunk 1]
```diff
@@ -553,6 +553,21 @@ def test_aware_datetime_week_view(self):
...
+    def test_incompatible_iso_week_format_view(self):
+        with self.assertRaisesMessage(
+            ValueError,
+            "ISO week directive '%V' is incompatible with the year directive '%Y'. "
+            "Use the ISO year '%G'."
+        ):
+            self.client.get('/dates/books/2008/week/39/incompatible_iso_week_format/')
+
+    def test_iso_week_format_view(self):
+        res = self.client.get('/dates/books/2008/week/39/iso_week_format/')
```
Comment (felixxm) [inline tests/generic_views/test_dates.py:565]:
```suggestion
        res = self.client.get('/dates/books/2008/week/40/iso_week_format/')
```

File: tests/generic_views/test_dates.py [hunk 2]
```diff
@@ -553,6 +553,21 @@ def test_aware_datetime_week_view(self):
...
+        self.assertEqual(res.status_code, 200)
+        self.assertTemplateUsed(res, 'generic_views/book_archive_week.html')
+        self.assertEqual(res.context['book_list'][0], Book.objects.get(pubdate=datetime.date(2008, 10, 1)))
+        self.assertEqual(res.context['week'], datetime.date(2008, 9, 28))
```
Comment (felixxm) [inline tests/generic_views/test_dates.py:569]:
```suggestion
        self.assertEqual(res.context['week'], datetime.date(2008, 9, 29))
```

File: tests/generic_views/test_dates.py [hunk 3]
```diff
@@ -553,6 +553,21 @@ def test_aware_datetime_week_view(self):
         res = self.client.get('/dates/booksignings/2008/week/13/')
         self.assertEqual(res.status_code, 200)
 
+    def test_incompatible_iso_week_format_view(self):
+        with self.assertRaisesMessage(
+            ValueError,
+            "ISO week directive '%V' is incompatible with the year directive '%Y'. "
+            "Use the ISO year '%G'."
+        ):
```
Comment (felixxm) [inline tests/generic_views/test_dates.py:561]:
```python
msg = ...
with self.assertRaisesMessage(ValueError, msg):
```

File: django/views/generic/dates.py [hunk 4]
```diff
@@ -485,15 +485,23 @@ def get_dated_items(self):
...
             ))
-        date = _date_from_string(year, self.get_year_format(),
+        year_format = self.get_year_format()
+        if week_format == '%V' and year_format != '%G':
+            raise ValueError(
+                "ISO week directive '%%V' is incompatible with the year "
+                "directive '%s'. Use the ISO year '%%G'." % (
+                    year_format
+                )
+            )
```
Comment (felixxm) [inline django/views/generic/dates.py:503]:
```python
raise ValueError( 
    "ISO week directive '%s' is incompatible with the year "
    "directive '%s'. Use the ISO year '%%G' instead." % (
        week_format, year_format,
    )   
)   
```

File: django/views/generic/dates.py [hunk 5]
```diff
@@ -220,7 +220,7 @@ def _get_weekday(self, date):
         week_format = self.get_week_format()
         if week_format == '%W':                 # week starts on Monday
             return date.weekday()
-        elif week_format == '%U':               # week starts on Sunday
+        elif week_format in ['%U', '%V']:       # week starts on Sunday
```
[the remark this answers is not in the document]
Comment (hramezani) [reply django/views/generic/dates.py]: Sorry, I Misunderstood it.

**Pull request merged**